aptos-move/framework/src/natives/lite_account.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::function_info::extract_function_info;
+use aptos_gas_schedule::gas_params::natives::aptos_framework::DISPATCHABLE_AUTHENTICATE_DISPATCH_BASE;
+use aptos_native_interface::{
+    RawSafeNative, SafeNativeBuilder, SafeNativeContext, SafeNativeError, SafeNativeResult,
+};
+use move_vm_runtime::native_functions::NativeFunction;
+use move_vm_types::{loaded_data::runtime_types::Type, values::Value};
+use smallvec::SmallVec;
+use std::collections::VecDeque;
+
+/***************************************************************************************************
+ * native fun dispatchable_authenticate
+ *
+ *   Directs control flow based on the last argument. We use the same native function implementation
+ *   for all dispatching native.
+ *   gas cost: a flat fee because we charged the loading of those modules previously.
+ *
+ **************************************************************************************************/
+pub(crate) fn native_dispatch(
+    context: &mut SafeNativeContext,
+    ty_args: Vec<Type>,
+    mut arguments: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    let (module_name, func_name) = extract_function_info(&mut arguments)?;
+    // Check if the module is already properly charged in this transaction.
+    if !context
+        .traversal_context()
+        .visited
+        .contains_key(&(module_name.address(), module_name.name()))
+    {
+        return Err(SafeNativeError::Abort { abort_code: 4 });
+    }
+
+    // Use Error to instruct the VM to perform a function call dispatch.
+    Err(SafeNativeError::FunctionDispatch {
```

Comment:
I think @runtian-zhou you should check this, I see we have
```rust
/// Load up a module and charge the module accordingly.
///
/// It is critical to invoke this function before calling FunctionDispatch to make sure the module loading
/// is charged properly, otherwise it would be a potential gas issue.
LoadModule { module_name: ModuleId },
```

Reply:
@runtian-zhou Could you have a look? I am not sure what's the issue here.